source/nvwave.py

```diff
@@ -223,14 +331,17 @@ def _feedUnbuffered(self, data, onDone=None):
 		with self._lock:
 			with self._waveout_lock:
 				self.open()
-				with self._global_waveout_lock:
-					winmm.waveOutPrepareHeader(self._waveout, LPWAVEHDR(whdr), sizeof(WAVEHDR))
 				try:
 					with self._global_waveout_lock:
+						winmm.waveOutPrepareHeader(self._waveout, LPWAVEHDR(whdr), sizeof(WAVEHDR))
 						winmm.waveOutWrite(self._waveout, LPWAVEHDR(whdr), sizeof(WAVEHDR))
-				except WindowsError as e:
-					self.close()
-					raise e
+				except WindowsError:
+					log.info("Error during feed. Resetting the device.")
+					self._close()  # don't try to call stop on a "broken" device.
+					self._setCurrentDevice(self._preferredDeviceName)
+					self.open()

```

Comment:
Technically this line is not needed as the following call to feed (and therefore _feedUnbuffered) will call self.open anyway?

Reply:
This is true, I wrote it like this just to make the intent clear. I toyed with the idea of removing the first call to open, however it is required to support "close when idle"